lib/veres-one-did-doc.js

```diff
@@ -0,0 +1,444 @@
+   * Only called when generating a new DID Doc (creates new keys, etc).
+   *
+   * @param [env]
+   * @param [passphrase]
+   * @param [keyType] {string}
+   * @param [didType] {string}
+   *
+   * @returns {Promise}
+   */
+  async init({env, passphrase, didType, keyType = constants.DEFAULT_KEY_TYPE}) {
+    const keyOptions = {injector: this.injector, type: keyType, passphrase};
+
+    // Generate a capabilityInvocation key, to base the DID URI on
+    const invokeKey = await LDKeyPair.generate(keyOptions);
+    const did = this.generateId({keyPair: invokeKey, didType, env});
+    this.doc.id = did;
+
+    // Generate an authentication key pair and suite
+    const authKey = await LDKeyPair.generate(keyOptions);
+    authKey.id = `${did}#authn-key-1`;
+    this.doc[constants.SUITES.authentication] = [
+      this.generateSuite({key: authKey, suiteId: `${did}#auth-suite-1`})
+    ];
+    this.keys[authKey.id] = authKey;
+
+    // Generate a capabilityDelegation key pair and suite
+    const delegateKey = await LDKeyPair.generate(keyOptions);
+    delegateKey.id = `${did}#ocap-delegate-key-1`;
+    this.doc[constants.SUITES.capabilityDelegation] = [
+      this.generateSuite({key: delegateKey, suiteId: `${did}#delegate-suite-1`})
+    ];
+    this.keys[delegateKey.id] = delegateKey;
+
+    // Generate a capabilityInvocation suite (from an earlier generated key)
+    invokeKey.id = `${did}#ocap-invoke-key-1`;
+    this.doc[constants.SUITES.capabilityInvocation] = [
+      this.generateSuite({key: invokeKey, suiteId: `${did}#invoke-suite-1`})
+    ];
+    this.keys[invokeKey.id] = invokeKey;
+  }
+
+  /**
+   * Generates a DID uri, either as a globally unique random string (uuid),
+   * or from a given key pair (in case of cryptonym type did).
+   *
+   * @param [keyPair] {LDKeyPair}
+   * @param [didType='nym'] {string} 'uuid' or 'nym'. If 'nym', a key pair
+   *   must also be passed in (to generate the did uri from).
+   * @param [env='dev'] {string}
+   *
+   * @returns {string} DID uri
+   */
+  generateId({keyPair, didType = 'nym', env = 'dev'}) {
+    if(didType === 'uuid') {
+      const prefix = (env === 'live') ? 'did:v1:' : 'did:v1:test:';
+      return (prefix + 'uuid:' + uuid()).replace(/-/g, '');
+    }
+
+    if(!keyPair) {
+      throw new Error('Cannot generate a cryptonym DID without a key');
+    }
+
+    // didType === 'nym'
+    return this.createCryptonymDid({keyPair, env});
+  }
+
+  generateSuite({key, suiteId}) {
+    const suiteType = key.type === 'Ed25519VerificationKey2018'
+      ? 'Ed25519SignatureCapabilityAuthorization2018'
+      : 'RsaSignatureCapabilityAuthorization2018';
+
+    return {
+      id: suiteId,
+      type: suiteType,
+      publicKey: [ key.publicNode({owner: this.id}) ]
+    };
+  }
+
+  suiteForId(suiteId) {
+    for(const suiteType in constants.SUITES) {
+      const suites = this.doc[suiteType];
+      const found = suites.find(s => s.id === suiteId);
+      if(found) {return found;}
```

Comment:
Got it, will change (re no one liner).

As far as throwing NotFoundError - I certainly can? Not seeing a pressing need for it though.

Reply:
So, this API is used on L392.  You don't test for return value which means that it might result in cannot find `publicKey` of undefined.  As a consumer of the API, would you like to debug that error or a meaningful NotFoundError that tells you that an invalid suite has been specified (or whatever)?

Reply:
Good point. I'll add that error.

Reply:
Ok, I fixed the one liner, and added the error in the calling code, downstream.